### DISCUSSION TITLE
Params in dag details are alphabetically ordered when querying the api

### DISCUSSION THREAD

**Discussion by shnups** [apache/airflow#27486] [Q&A]
Where querying the api for dag details (`/api/v1/dags/{dag_id}/details`), I get the `params` field order alphabetically. Tested on airflow 2.2.2 and 2.4.2

It surprised me as we are loosing the original order of params defined in the dag and I tend to think applying alphabetically ordering by default doesn't serve a specific purpose. However, in my opinion, keeping the order from the dag does. 

When triggering a manual dag run with config via the UI, params are presented in the order defined in the dag. This makes me think this an api-only problem (maybe just related to the serialization of response).

Before creating a bug and/or investigating the code further, I was wondering if this was a voluntary decision or not. I'm building an internal tool at work that launch dag programatically and params order is important. What course of actions would you recommand?

Thanks for your time helping out :)

**Comment by potiuk** [comment]
You can always raise PR and see if it wll be well received. I think for params it might be quite acceptable, I don't think there is a specifi "contract" about the sequence. But just a warning - unless you spend your time on implementing it, if you open  a feature, it might be marked as good first issue and there will be no-one to take an interest in implementing it (this is how open-source works) - so opening a PR with draft proposal and leading it to completion is the best way. However it might be abit complex because you need to keep the params sequence across serialization and deserialization. But I think if you implement it, add testing and it will be reasonable code there should be no objects (but this is just my personal view others might have different opinion and the best thing is to try).
